recaptchaenterprise/samples/snippets/create_mfa_assessment.py

```diff
@@ -0,0 +1,101 @@
+               To get the token, integrate the recaptchaSiteKey with frontend. See,
+               https://cloud.google.com/recaptcha-enterprise/docs/instrument-web-pages#frontend_integration_score
+        recaptcha_action: The action name corresponding to the token.
+        hashed_account_id: Create hashedAccountId from user identifier.
+                           It's a one-way hash of the user identifier: HMAC SHA 256 + salt
+        email: Email id of the user to trigger the MFA challenge.
+        phone_number: Phone number of the user to trigger the MFA challenge.
+    """
+
+    client = recaptchaenterprise_v1.RecaptchaEnterpriseServiceClient()
+
+    # Set the properties of the event to be tracked.
+    event = recaptchaenterprise_v1.Event()
+    event.site_key = recaptcha_site_key
+    event.token = token
+    event.hashed_account_id = hashed_account_id
+
+    # Set the email address and the phone number to trigger/ verify the MFA challenge.
+    endpoint_verification_info = recaptchaenterprise_v1.EndpointVerificationInfo()
+    endpoint_verification_info.email_address = email
+    endpoint_verification_info.phone_number = phone_number
+
+    account_verification_info = recaptchaenterprise_v1.AccountVerificationInfo()
+    account_verification_info.endpoints = endpoint_verification_info
+
+    assessment = recaptchaenterprise_v1.Assessment()
+    assessment.event = event
+    assessment.account_verification = account_verification_info
+
+    project_name = f"projects/{project_id}"
+
+    # Build the assessment request.
+    request = recaptchaenterprise_v1.CreateAssessmentRequest()
+    request.assessment = assessment
+    request.parent = project_name
+
+    # Check integrity of the response.
+    response = client.create_assessment(request)
+    if not verify_response_integrity(response, recaptcha_action):
+        raise Exception("Failed to verify token integrity.")
+
+    result = response.account_verification.latest_verification_result
+    # If the result is unspecified, send the request token to trigger MFA in the client.
+    # You can choose to send both the email and phone number's request token.
+    if result == recaptchaenterprise_v1.types.AccountVerificationInfo.Result.RESULT_UNSPECIFIED:
+        # Send the request token for assessment. The token is valid for 15 minutes.
+        print("Result unspecified. Triggering MFA challenge.")
+        # print(response.account_verification.endpoints[0].request_token)
+
+    # If the result is not unspecified, return the result.
+    print(f"MFA result: {result}")
+
+
+def verify_response_integrity(response: Assessment, recaptcha_action: str) -> bool:
+    # Check if the token is valid.
+    if not response.token_properties.valid:
+        print(
+            "The CreateAssessment call failed because the token was "
+            + "invalid for for the following reasons: "
+            + str(response.token_properties.invalid_reason)
```

Comment:
Approving - but I'm wondering if there's a way to use an f-string here while staying under the line length limits.